utils/CircuitCheck/UnitaryBuilder.cpp

```diff
@@ -34,12 +34,16 @@ LogicalResult UnitaryBuilder::build(func::FuncOp func) {
     if (auto optor = dyn_cast<quake::OperatorInterface>(op)) {
       optor.getOperatorMatrix(matrix);
       // If the operator couldn't produce a matrix, stop the walk.
-      if (matrix.empty())
+      if (matrix.empty()) {
+        optor.emitError("Couldn't produce matrix.");
```

Comment:
If you use `emitOpError` I believe it gives more information about which Op is the one failing in the diagnostic.

Reply:
Good point. I actually didn't know about `emittOpError`! Thanks.